osprey_flask_app/routes.py

```diff
@@ -44,17 +46,51 @@ def osprey_route():
     """
     args = request.args
     arg_dict = create_full_arg_dict(args)
-    outpath = run_full_rvic(arg_dict)
+    validate_response = validate_inputs(arg_dict)
+    if validate_response.status_code == 400:
+        return validate_response
+
+    rvic_thread = threading.Thread(
```

Comment:
Again, this is a great start for getting a prototype up. Though I have concerns about using the `thread_id` as the job management id. This basically means that we wouldn't be able to scale horizontally. For example, imagine that we had 4 instances of this service and we were round robining between them. Depending on how the thread_ids are generated, there would be a chance of collision... where multiple services generate the same thread_id. Then when subsequent requests come in for that id, which does it actually go to?

We don't need to worry about that for this PR, but please create a GH issue that outlines this problem.

Reply:
I also *was* going to say that I was concerned about the use of threads to do the compute (again, how would this scale?). But then I realized that the thread isn't really doing any *work*; it's just making a WPS request and then waiting around for the answer. So that should work. Have you tried it on a long running request?